File: contrib/sdk-bindings/kotlin/demo/app/src/test/kotlin/org/example/TestEntrypoints.kt
/* SPDX-FileCopyrightText: 2025 Functori <contact@functori.com>
 *
 * SPDX-License-Identifier: MIT
 */

package org.example.tezos

import kotlin.test.Test
import kotlin.test.assertEquals

class EntrypointsTest {

    @Test fun buildEntrypoint() {
        val entrypoint = Entrypoint("")
        assertEquals(entrypoint.toString(), "default")
    }
}

File: contrib/sdk-bindings/swift/Tests/TestEntrypoints.swift
// SPDX-FileCopyrightText: 2025 Functori <contact@functori.com>
//
// SPDX-License-Identifier: MIT

import XCTest
import Tezos

class TestEntrypoints: XCTestCase {

    public static var allTests = [
      ("buildEntrypoint", buildEntrypoint),
    ]

    func buildEntrypoint() {
        let rawEntrypoint = "my_custom_entrypoint"
        let entrypoint = try! Entrypoint(name: rawEntrypoint)
        XCTAssertEqual(String(describing: entrypoint), rawEntrypoint)
    }
}
